### PR TITLE
Fix a bunch of errors found by ESLint and TypeScript

### PR DESCRIPTION
**Pull request by AlCalzone**

### CODE REVIEW

File: lib/setup/setupInstall.js [hunk 1]
```diff
@@ -1444,8 +1444,8 @@ function Install(options) {
             adapter, adapter + '.admin'
         ]) {
             try {
-                const obj = await delObjectAsync(objId);
-                obj && console.log(`host.${hostname} object ${objId} deleted`);
+                await delObjectAsync(objId);
```
Comment (AlCalzone) [inline lib/setup/setupInstall.js]: @Apollon77 here, it is used with the assumption that `delObjectAsync(...)` resolves to an object, which it doesn't

File: lib/states/statesInRedis.js [hunk 2]
```diff
@@ -30,7 +30,6 @@ function bufferJsonDecoder(key, value) {
 class StateRedis {
 
     constructor(settings) {
-        const originalSettings = settings;
```
Comment (AlCalzone) [inline lib/states/statesInRedis.js]: unused

File: lib/states/statesInRedis.js [hunk 3]
```diff
@@ -609,14 +608,14 @@ class StateRedis {
             const result = [];
 
             obj = obj || [];
-            obj.forEach((state, i) => {
+            obj.forEach(state => {
```
Comment (AlCalzone) [inline lib/states/statesInRedis.js]: i is unused

File: lib/tools.js [hunk 4]
```diff
@@ -269,7 +269,7 @@ function getMac(callback) {
             let result = null;
 
             while (true) {
-                match = macRegex.exec(stdout, {windowsHide: true});
```
Comment (AlCalzone) [inline lib/tools.js]: that is just nonsense

File: lib/tools.js [hunk 5]
```diff
@@ -1168,7 +1165,7 @@ function getCertificateInfo(cert) {
  *
  * @alias generateDefaultCertificates
  * @memberof Tools
- * @returns {defaultPrivate: any[],defaultPublic: any[]}
+ * @returns {{ defaultPrivate: any[];defaultPublic: any[] }}
```
Comment (AlCalzone) [inline lib/tools.js]: To declare an object return type in JSDoc, you need two pairs of braces. One for the comment, one for the type

File: main.js [hunk 6]
```diff
@@ -3029,9 +3029,9 @@ function startInstance(id, wakeUp) {
                     ) {
                         states.setState(id + '.sigKill', {val: 0, ack: false, from: hostObjectPrefix}); // set to 0 to stop any pot. already running instances, especially broken compactModes
 
+                        const _instance = (instance && instance._id && instance.common) ? instance._id.split('.').pop() || 0 : 0;
```
Comment (AlCalzone) [inline main.js]: `_instance` was accessed in the else branch where it is not defined

File: lib/tools.js (hunk 4 above)
Comment (Apollon77) [reply lib/tools.js]: ;-)

File: main.js (hunk 6 above)
Comment (Apollon77) [reply main.js]: good catch

**Comment by Apollon77** [comment]
Thank you!

**Pull request merged**